### PR TITLE
Don't automatically deploy a commit without it's revert

### PR DESCRIPTION
**Pull request by fw42**
Alternative implementation of https://github.com/Shopify/shipit-engine/pull/669.

@kmcphillips @casperisfine, thoughts?

### CODE REVIEW

**Comment by casperisfine** [comment]
It's better, but it still has a couple issues:

  - The revert detection is quite fragile
  - There is no way for a human to overrule it.
  - It requires relatively heavy queries and computations to generate the `stacks#show` page.

I'll believe I'll go with something along the line of https://github.com/Shopify/shipit-engine/issues/270

I'm thinking adding a boolean `locked` attribute on `Commit`. When a revert reaches master, if the commit it reverts is not deployed yet, then all the range will get the `locked` property.

In additions, on the `stacks#show` page, each undeployed commit will get a button to toggle locks on commits. That way we get both automated security, and fine grained manual management.

File: app/models/shipit/commit.rb [hunk 1]
```diff
@@ -109,6 +109,21 @@ def deployable?
       success? || stack.ignore_ci?
     end
 
+    def safe_to_deploy?
```
Comment (fw42) [inline app/models/shipit/commit.rb]: maybe this should say "safe_to_deploy_automatically?"

**Comment by fw42** [comment]
> I'll believe I'll go with something along the line of #270

ok, fair enough. I'll leave this alone for now then and give you some time to experiment with your ideas. Let me know if you need me to revisit this.

**Comment by casperisfine** [comment]
👌

**Pull request closed without merging**